Source/Core/Cartesian3.js

```diff
@@ -341,15 +318,9 @@ define([
      */
     Cartesian3.maximumByComponent = function(first, second, result) {
         //>>includeStart('debug', pragmas.debug);
-        if (!defined(first)) {
-            throw new DeveloperError('first is required.');
-        }
-        if (!defined(second)) {
-            throw new DeveloperError('second is required.');
-        }
-        if (!defined(result)) {
-            throw new DeveloperError('result is required.');
-        }
+        Check.defined(first, 'first');
```

Comment:
I wasn't quite sure if we were actually going to use the specific type checks, before knowing about the performance, but I'll go ahead and replace these now.